### PR TITLE
Add MimeType class.

### PR DESCRIPTION
**Pull request by ADmad**
It allows users to easily add additional extension to mime-type mappings.

Refs #18004.

### CODE REVIEW

File: src/Http/Response.php [hunk 1]
```diff
@@ -684,7 +441,7 @@ public function getReasonPhrase(): string
      */
     public function setTypeMap(string $type, array|string $mimeType): void
     {
-        $this->_mimeTypes[$type] = $mimeType;
```
Comment (markstory) [inline src/Http/Response.php:687]: I'll add this to the migration notes. While we haven't changed any public APIs we could trip up application code with this change.

File: src/Http/Response.php [hunk 2]
```diff
@@ -1358,31 +1116,19 @@ public function withFile(string $path, array $options = []): static
...
         $fileSize = $file->getSize();
         if ($options['download']) {
-            $agent = (string)env('HTTP_USER_AGENT');
-
-            $contentType = null;
-            if ($agent && preg_match('%Opera([/ ])([0-9].[0-9]{1,2})%', $agent)) {
-                $contentType = 'application/octet-stream';
-            } elseif ($agent && preg_match('/MSIE ([0-9].[0-9]{1,2})/', $agent)) {
-                $contentType = 'application/force-download';
-            }
```
Comment (markstory) [inline src/Http/Response.php:1381]: Nice to see these workarounds for legacy browsers leaving as well.

**Review by markstory — APPROVED** [review]

File: src/Http/Response.php [hunk 3]
```diff
@@ -764,16 +528,10 @@ public function getMimeType(string $alias): array|string|false
...
     {
         if (is_array($ctype)) {
-            return array_map($this->mapType(...), $ctype);
-        }
-
-        foreach ($this->_mimeTypes as $alias => $types) {
-            if (in_array($ctype, (array)$types, true)) {
-                return $alias;
-            }
+            return array_map('\Cake\Http\MimeType::getExtension', $ctype);
```
Comment (dereuromark) [inline src/Http/Response.php:531]: We might be missing a test for this
```
I get
TypeError: Cake\Http\MimeType::getExtension(): Argument #1 ($mimeType) must be of type string, array given

/data/work/git/cakephp-shim/vendor/cakephp/cakephp/src/Http/MimeType.php:352
/data/work/git/cakephp-shim/vendor/cakephp/cakephp/src/Http/Response.php:531
```

Comment (dereuromark) [reply src/Http/Response.php:531]:
```php
$content = new ContentTypeNegotiation();
$accept = $content->parseAccept($request);
$accepts = $response->mapType($accept);		
```
and apparently a result for parseAccept as
```
[
  '1.0' => [
    (int) 0 => 'text/csv'
  ],
  '0.8' => [
    (int) 0 => 'application/json'
  ],
  '0.7' => [
    (int) 0 => 'application/xml'
  ]
]
```
worked before

and now breaks here as the array_map() is not handling this anymore.

**Pull request merged**